Library/Homebrew/test/language/java_spec.rb

```diff
@@ -32,24 +42,24 @@
   describe "::java_home_env" do
     it "returns java_home path if version specified" do
       java_home_env = described_class.java_home_env("1.8+")
-      expect(java_home_env[:JAVA_HOME]).to eql(f.opt_libexec.to_s)
+      expect(java_home_env[:JAVA_HOME]).to include(f.opt_libexec.to_s)
     end
 
     it "returns java_home path if version is not specified" do
       java_home_env = described_class.java_home_env
-      expect(java_home_env[:JAVA_HOME]).to eql(f.opt_libexec.to_s)
+      expect(java_home_env[:JAVA_HOME]).to include(f.opt_libexec.to_s)
     end
   end
 
   describe "::overridable_java_home_env" do
     it "returns java_home path if version specified" do
       overridable_java_home_env = described_class.overridable_java_home_env("1.8+")
-      expect(overridable_java_home_env[:JAVA_HOME]).to eql("${JAVA_HOME:-#{f.opt_libexec}}")
+      expect(overridable_java_home_env[:JAVA_HOME]).to include(f.opt_libexec.to_s)
     end
 
     it "returns java_home path if version is not specified" do
       overridable_java_home_env = described_class.overridable_java_home_env
-      expect(overridable_java_home_env[:JAVA_HOME]).to eql("${JAVA_HOME:-#{f.opt_libexec}}")
+      expect(overridable_java_home_env[:JAVA_HOME]).to include(f.opt_libexec.to_s)
```

Comment:
I think it'd be good to keep this as `eql` and add separate macOS/Linux tests if needed.